homeassistant/components/sensor/dsmr.py

```diff
@@ -351,7 +352,8 @@ def state(self):
                 # Recalculate the rate
                 diff = current_reading - self._previous_reading
                 timediff = timestamp - self._previous_timestamp
-                self._state = diff / timediff * 3600
+                total_seconds = timediff.total_seconds()
+                self._state = round(diff / Decimal(total_seconds) * 3600, 3)
```

Comment:
Why do we need `Decimal()` here?

Reply:
Because dividing a decimal by a float is not acceptable to python either apparently.

Reply:
Ah okay. I seem unable to remember that `total_seconds()` is float and so I tested with integer division.

You could avoid the import by using `float(diff) / total_seconds` but I guess the current solution is okay.

Reply:
Changed it :)

Reply:
Little bit of nitpickery, but I think the `Decimal()` should only be around the `total_seconds`. It probably won't make much difference here but you're doing floating point calculations only to convert it to decimal after. So you lose precision that way :)

Reply:
@WoLpH It sounds like you have misread.

Reply:
@amelchio oops... you're right, one of my github extensions drew a character limit line right on top of the `)`. My bad ;)